lib/dnsimple/client.rb

```diff
@@ -147,16 +151,11 @@ def execute(method, path, data, options = {})
     #
     # @param  [String] method The HTTP method
     # @param  [String] path The path, relative to {#base_url}
-    # @param  [Hash] options Query and header params for request
+    # @param  [Hash] data The body for the request
+    # @param  [Hash] options The query and header params for the request
     # @return [HTTParty::Response]
     def request(method, path, data, options = {})
-      if data.is_a?(Hash)
-        options[:query]   = data.delete(:query)   if data.key?(:query)
-        options[:headers] = data.delete(:headers) if data.key?(:headers)
-      end
-      if !data.empty?
-        options[:body] = data
-      end
+      options[:body] = data if !data.empty?
```

Comment:
Also this makes me think that we should initialize the argument to be `nil` for every method... Otherwise empty payloads would require passing in a `nil` which doesn't look too good.